packages/lending/src/adapter.ts

```diff
@@ -185,6 +185,299 @@ export class Adapter extends common.Web3Toolkit {
+      logics: [],
+    };
+    try {
+      // 1. ---------- validate params ----------
+      if (collateralAmount === '0') throw new OperationError('collateralAmount', 'ZERO_AMOUNT');
+
+      const srcCollateral = portfolio.findSupply(collateralToken);
+      if (!srcCollateral) throw new OperationError('collateralToken', 'UNSUPPORTED_TOKEN');
+
+      const destBorrow = portfolio.findBorrow(debtToken);
+      if (!destBorrow) throw new OperationError('debtToken', 'UNSUPPORTED_TOKEN');
+
+      const supplyAmountDelta = new common.TokenAmount(collateralToken.wrapped, collateralAmount).sub(
+        srcCollateral.balance
+      );
+      if (!srcCollateral.validateSupplyCap(supplyAmountDelta.amount)) {
+        throw new OperationError('collateralAmount', 'SUPPLY_CAP_EXCEEDED');
+      }
+
+      // 2. ---------- zap swap ----------
+      let supplyInput = new common.TokenAmount(collateralToken.wrapped, '0');
+      if (zapAmount !== '0' && !zapToken.wrapped.is(collateralToken.wrapped)) {
+        const swapper = this.findSwapper([zapToken, collateralToken]);
+        let swapQuotation: SwapperQuoteFields;
+        try {
+          swapQuotation = await swapper.quote({
+            input: new common.TokenAmount(zapToken, zapAmount),
+            tokenOut: collateralToken.wrapped,
+            slippage,
+          });
+        } catch {
+          throw new OperationError('zapAmount', 'NO_ROUTE_FOUND_OR_PRICE_IMPACT_TOO_HIGH');
+        }
+        const swapTokenLogic = swapper.newSwapTokenLogic(swapQuotation);
+        output.logics.push(swapTokenLogic);
+        // 2-1. the supply amount is the swap quotation output
+        supplyInput = swapQuotation.output;
+        output.afterPortfolio.supply(supplyInput.token, supplyInput.amount);
+      } else if (zapToken.wrapped.is(collateralToken.wrapped)) {
+        supplyInput = new common.TokenAmount(zapToken, zapAmount);
+        output.afterPortfolio.supply(supplyInput.token, supplyInput.amount);
+      }
+
+      // 3. leverage by collateral
+      const leverageAmount = supplyAmountDelta.sub(supplyInput.amount).amount;
+      const leverageOutput = await this.leverageByCollateral({
+        account,
+        portfolio: output.afterPortfolio,
+        srcToken: supplyInput.token,
+        srcAmount: leverageAmount,
+        destToken: debtToken,
+      });
+
+      if (leverageOutput.error) throw leverageOutput.error;
+
+      // 3-1. find supply logic and add zap supply amount
+      if (!collateralToken.wrapped.is(debtToken.wrapped)) {
+        leverageOutput.logics[2].fields.input.amount = supplyInput.add(
+          leverageOutput.logics[2].fields.input.amount
+        ).amount;
+        leverageOutput.logics[2].fields.balanceBps = common.BPS_BASE;
+      } else {
+        leverageOutput.logics[1].fields.input.amount = supplyInput.add(
+          leverageOutput.logics[1].fields.input.amount
+        ).amount;
+        leverageOutput.logics[1].fields.balanceBps = common.BPS_BASE;
+      }
```

Comment:
Can we swap the `if` and `else` blocks to eliminate `!` and keep consistency with other functions?

And we can simply this code by set index such as 
```
const logicIndex = collateralToken.wrapped.is(debtToken.wrapped) ? 1 : 2;